src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -121,6 +122,19 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
                 $firstResult = (int) $after;
                 $firstResult = false === $after ? $firstResult : ++$firstResult;
             }
+            if (isset($collectionArgs[$resourceClass]['last'])) {
```

Comment:
Are you doing this for the case the `before` is missing? I think we shouldn't manage this case. The `last` should be only managed when a `before` is present.
And worse, you are doing an extra query inside a building of a query.

Reply:
Yeah, that's kinda hacky. It's for the case that before isn't set, so you can get the last items in the correct order. I have to think about another way

Reply:
I didn't come up with an idea so far :-/ How should I we this?

Reply:
I think doing backwards pagination in SQL is tricky and you need to know the total rows the first time in order to do it.
The current implementation needs to be changed though because `getArrayResult` can be very big and I think it can eat all the memory. You need to use an SQL count.
The Doctrine Paginator is doing it anyway when you get the total items (see its method `getCountQuery`). You need to do something like this.